vars/influxWriteData.groovy

```diff
@@ -2,50 +2,58 @@ import com.sap.piper.ConfigurationLoader
 import com.sap.piper.ConfigurationMerger
 import com.sap.piper.JsonUtils
 
-def call(Map parameters = [:]) {
+import groovy.transform.Field
 
-    def stepName = 'influxWriteData'
+@Field def STEP_NAME = 'influxWriteData'
 
-    handlePipelineStepErrors (stepName: stepName, stepParameters: parameters, allowBuildFailure: true) {
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field Set STEP_CONFIG_KEYS = [
+    'influxServer',
+    'influxPrefix'
+]
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS.plus([
+    'artifactVersion'
+])
 
-        def script = parameters.script
-        if (script == null)
-            script = [commonPipelineEnvironment: commonPipelineEnvironment]
+def call(Map parameters = [:]) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters, allowBuildFailure: true) {
+        def script = parameters.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
 
-        prepareDefaultValues script: script
+        // load default & individual configuration
+        Map configuration = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin([
+                artifactVersion: commonPipelineEnvironment.getArtifactVersion()
+            ])
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
 
-        Set parameterKeys = [
-            'artifactVersion',
-            'influxServer',
-            'influxPrefix'
-        ]
         Map pipelineDataMap = [
             artifactVersion: commonPipelineEnvironment.getArtifactVersion()
         ]
-        Set stepConfigurationKeys = [
-            'influxServer',
-            'influxPrefix'
-        ]
 
-        Map configuration = ConfigurationMerger.merge(script, stepName, parameters, parameterKeys, pipelineDataMap, stepConfigurationKeys)
+        Map configuration = ConfigurationMerger.merge(script, STEP_NAME, parameters, parameterKeys, pipelineDataMap, stepConfigurationKeys)
 
         def artifactVersion = configuration.artifactVersion
         if (!artifactVersion)  {
             //this takes care that terminated builds due to milestone-locking do not cause an error
-            echo "[${stepName}] no artifact version available -> exiting writeInflux without writing data"
+            echo "[${STEP_NAME}] no artifact version available -> exiting writeInflux without writing data"
             return
         }
 
         def influxServer = configuration.influxServer
         def influxPrefix = configuration.influxPrefix
 
-        echo """[${stepName}]----------------------------------------------------------
+        echo """[${STEP_NAME}]----------------------------------------------------------
```

Comment:
Be focused: PR is about adjusting the configuration, it is not about STEP_NAME, it is not above providing the script in a more groovy like way. Both is fine, but would be worth a dedicated pull request. This simplifies and speeds up reviews.

Reply:
removed unfocused changes